didcomm/message.py

```diff
@@ -37,21 +37,46 @@ class GenericMessage(Generic[T]):
     custom_headers: Optional[List[Header]] = None
 
     def as_dict(self) -> dict:
-        d = dataclasses.asdict(self)
-        for k in set(d.keys()):
-            if d[k] is None:
-                del d[k]
+        d = dataclass_to_dict(self)
+
         if "frm" in d:
             d["from"] = d["frm"]
             del d["frm"]
+
+        d["typ"] = DIDCommMessageTypes.PLAINTEXT.value
+
+        if self.attachments:
+            d["attachments"] = [a.as_dict() for a in self.attachments]
+        if self.from_prior:
+            d["from_prior"] = self.from_prior.as_dict()
+
         return d
 
     @staticmethod
     def from_dict(d: dict) -> Message:
         if "from" in d:
             d["frm"] = d["from"]
             del d["from"]
-        return Message(**d)
+        del d["typ"]
```

Comment:
not clear place: don't we consider another (non plaintext) application types (e.g. `application/didcomm-encrypted+json`) for such a generic type? Possibly makes sense to rename the class to something like `PlaintextGenericMessage`.